# **TASA INTERNA DE RETORNO (TIR)**

La Tasa Interna de Retorno (TIR) es un indicador financiero que mide la rentabilidad de un proyecto con base en una inversión inicial y en unos ingresos periódicos proyectados traídos a valor presente neto, de modo que este sea igual a cero.

En otras palabras, la TIR representa la tasa de descuento que iguala el valor presente de los flujos de caja futuros de una inversión con el costo inicial, haciendo que el Valor Presente Neto (VPN) sea igual a cero

La Tasa Interna de Retorno (TIR) mide la rentabilidad promedio anual de una inversión o proyecto, expresada como un porcentaje.

Específicamente, indica la tasa de rendimiento que los flujos de caja futuros generan sobre el capital invertido, considerando el valor del dinero en el tiempo.



## Importancia de la (TIR)

La TIR es importante permite determinar si un proyecto es rentable comparándola con el costo de capital a una tasa mínima de retorno requerida. Si la TIR supera esa tasa, el proyecto es financieramente viable; si es menor, no lo es.

Además, en cierto modo, mide el punto de equilibrio financiero, donde los ingresos descontados a valor presente igualan la inversión inicial.




## Fórmula para calcular la tasa interna de retorno.

La fórmula para calcular la TIR es la siguiente:

$$\sum_{t=0}^{n} \frac{C_t}{(1 + \text{TIR})^t} = 0 $$

Las variables son las siguientes:  

- $C_t$ es el flujo de caja en el período $t$.  
- $n$ es el número total de períodos.  
- $\text{TIR}$ es la tasa interna de retorno a determinar.
- $t$ representa cada período, desde $t=0$ hasta $t=n$.

 ### COMO CALCULAR LA **TIR** EN EXCEL
![TIREXCEL](https://www.sage.com/es-es/blog/wp-content/uploads/sites/8/2021/11/FORMULA-1.jpg)




 ## Cómo interpretar la tasa interna de retorno.

 La TIR puede tener tres resultados posibles:
- Mayor que cero.
- Igual a cero.
- Menor que cero.

***TIR mayor que cero:*** Si la TIR es mayor que cero, significa que el proyecto es rentable, ya que los flujos de efectivo generados superan la inversión inicial. Sin embargo, la magnitud de la rentabilidad depende de qué tan lejos está la TIR de cero y de las expectativas del inversionista. Por ejemplo, una TIR de 5% (0,05) podría considerarse baja, mientras que una TIR de 30% (0,30) generalmente sería atractiva, dependiendo de las metas del inversionista.

***TIR igual a cero:*** Si la TIR es igual a cero, significa que el proyecto genera un retorno justo suficiente para cubrir el costo de la inversión, pero no genera valor adicional.

***TIR menor que cero:*** Si la TIR es menor que cero, el proyecto no es rentable, ya que los flujos de efectivo no son suficientes para recuperar la inversión inicial.

[LINKREFERENCIA](https://www.gerencie.com/tir.html)


### Ejercicio: Cálculo de la Tasa Interna de Retorno (TIR) utilizando la fórmula cuadrática

Una empresa está evaluando la viabilidad de un proyecto de inversión que requiere un desembolso inicial y genera flujos de efectivo en los años siguientes. Se desea calcular la Tasa Interna de Retorno (TIR) del proyecto utilizando un modelo matemático basado en la ecuación cuadrática.

## Pasos a seguir:

### 1. Definir los flujos de caja

- Se establece la inversión inicial como un valor negativo, ya que representa un desembolso.
- Se *asignan* los valores de los flujos de caja proyectados para los años siguientes.

In [ ]:

inversion_inicial = -70
Flujoano_1 = 15
Flujoano_2 = 60

### 2. Calcular el discriminante de la ecuación cuadrática

- Se aplica la fórmula del discriminante:

$$
Discriminante = b^2 - 4ac
$$

- Si el discriminante es negativo, no hay soluciones reales.

In [ ]:
discriminante = (Flujoano_1) ** 2 - 4 * (inversion_inicial) * (Flujoano_2)

### 3. Verificar si existen soluciones reales
- Si el discriminante es negativo, se indica que no hay soluciones reales y si el discriminante es positivo o cero, se procede con el cálculo de la TIR.
- Se usa la ecuación:

$$
x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}
$$


In [ ]:
if discriminante >= 0:  # Verificar si hay soluciones reales
 tir_suma = (-Flujoano_1 + (discriminante) ** 0.5) / (2 * (inversion_inicial))
 tir_resta = (-Flujoano_1 - (discriminante) ** 0.5) / (2 * (inversion_inicial))

 print("SUMA : ", tir_suma)
 print("RESTA :",tir_resta)

SUMA :  -0.8248563107642777
RESTA : 1.0391420250499919


### 5. Verificar cuál de las soluciones es válida

- Se selecciona la solución que sea positiva, ya que una TIR negativa no es financieramente viable.
- Se convierte el valor a porcentaje restando 1 ( pues $𝑥=1+𝑇𝐼𝑅)$


In [ ]:
if tir_suma > 0:
    print(f"La Tasa Interna de Retorno (TIR) es: {(tir_suma - 1) * 100}%")
elif tir_resta > 0:
    print(f"La Tasa Interna de Retorno (TIR) es: {(tir_resta - 1) * 100}%")
else:
    print("No se encontró una TIR válida (ambas soluciones son no positivas).")



La Tasa Interna de Retorno (TIR) es: 3.914202504999187%


# CODIGO CORRIDO

In [ ]:
# Definir los flujos de caja
inversion_inicial = -70  # Inversión inicial (negativa porque es un desembolso)
Flujoano_1 = 15
Flujoano_2 = 60

# Calcular las posibles TIR usando la fórmula cuadrática
discriminante = (Flujoano_1) ** 2 - 4 * (inversion_inicial) * (Flujoano_2)

if discriminante >= 0:  # Verificar si hay soluciones reales
    tir_suma = (-Flujoano_1 + (discriminante) ** 0.5) / (2 * (inversion_inicial))
    tir_resta = (-Flujoano_1 - (discriminante) ** 0.5) / (2 * (inversion_inicial))

    print("SUMA :",tir_suma)
    print("RESTA :",tir_resta)

    # Validar cuál de las dos soluciones es válida (debe estar entre 0 y 1)
    if tir_suma > 0:
        print(f"La Tasa Interna de Retorno (TIR) es: {(tir_suma-1) * 100}%")
    elif tir_resta > 0:
        print(f"La Tasa Interna de Retorno (TIR) es: {(tir_resta-1)* 100}%")
    else:
        print("No se encontró una TIR válida (ambas soluciones son no positivas).")
else:
    print("No hay soluciones reales para la TIR (discriminante negativo).")


SUMA : -0.8248563107642777
RESTA : 1.0391420250499919
La Tasa Interna de Retorno (TIR) es: 3.914202504999187%




## Interpretación del resultado:
La Tasa Interna de Retorno (TIR) es aproximadamente 3.91%.
Esto significa que, si descontamos los flujos de caja a una tasa del 3.91%, el Valor Presente Neto (VPN) del proyecto sería cero.
En términos financieros, el proyecto es rentable si la tasa de descuento del mercado es menor a 3.91%.
Si la tasa de descuento del mercado (costo del capital) es mayor a 3.91%, el proyecto no sería viable, ya que el costo de oportunidad sería demasiado alto.

### Conclusión:
Si buscas rentabilidad, compara esta TIR con la tasa de descuento de la empresa o del mercado para tomar una decisión informada.